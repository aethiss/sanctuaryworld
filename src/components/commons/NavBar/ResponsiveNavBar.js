import React from 'react'
import PropTypes from 'prop-types'

// Next
import Link from 'next/link'

// Material-Ui
import { makeStyles } from '@material-ui/core/styles'
import AppBar from '@material-ui/core/AppBar'
import Toolbar from '@material-ui/core/Toolbar'
import IconButton from '@material-ui/core/IconButton'
import Typography from '@material-ui/core/Typography'
// import Badge from '@material-ui/core/Badge'
import MenuItem from '@material-ui/core/MenuItem'
import Menu from '@material-ui/core/Menu'
import MenuIcon from '@material-ui/icons/Menu'
import AccountCircle from '@material-ui/icons/AccountCircle'
// import MailIcon from '@material-ui/icons/Mail'
// import NotificationsIcon from '@material-ui/icons/Notifications'
import MoreIcon from '@material-ui/icons/MoreVert'

const useStyles = makeStyles((theme) => ({
  grow: {
    flexGrow: 1,
  },
  appBar: {
    backgroundColor: '#181717',
    backgroundImage: 'url("/images/background/bg-content-top.jpg")',
  },
  divider: {
    height: '32px',
    backgroundImage: 'url("/images/background/heading-divider-top.png")',
    backgroundRepeat: 'no-repeat',
    backgroundPosition: 'center bottom',
    borderTop: '1px solid #171E21',
  },
  menuButton: {
    marginRight: theme.spacing(2),
  },
  growCenter: {
    flexGrow: 1,
    // paddingLeft: '4%',
  },
  title: {
    textAlign: 'center',
    fontFamily: 'Old Fenris,Volkhov,Vollkorn,Times New Roman,serif',
    textTransform: 'uppercase',
    display: 'block',
    [theme.breakpoints.up('sm')]: {
      display: 'block',
    },
  },
  icon: {
    width: '25px',
    position: 'relative',
    top: '5px',
    right: '10px',
    display: 'none',
    [theme.breakpoints.up('md')]: {
      display: 'inline',
    },
  },
  sectionDesktop: {
    display: 'none',
    [theme.breakpoints.up('md')]: {
      display: 'flex',
    },
  },
  sectionMobile: {
    display: 'flex',
    [theme.breakpoints.up('md')]: {
      display: 'none',
    },
  },
}))

const ResponsiveNavBar = ({ openMenu }) => {
  const classes = useStyles()
  const [menuMoreAnchorEl, setMenuMoreAnchorEl] = React.useState(null)

  const isMobileMenuOpen = Boolean(menuMoreAnchorEl)

  const handleMenuClose = () => {
    setMenuMoreAnchorEl(null)
  }

  const handleMenuOpen = (event) => {
    setMenuMoreAnchorEl(event.currentTarget)
  }

  const menuId = 'primary-search-account-menu'

  const mobileMenuId = 'primary-search-account-menu-mobile'
  const renderMobileMenu = (
    <>
      <Menu
        id={mobileMenuId}
        anchorEl={menuMoreAnchorEl}
        anchorOrigin={{
          vertical: 'top',
          horizontal: 'right',
        }}
        keepMounted
        transformOrigin={{
          vertical: 'top',
          horizontal: 'right',
        }}
        open={isMobileMenuOpen}
        onClose={handleMenuClose}
      >
        <MenuItem onClick={handleMenuClose}>My account</MenuItem>
        <MenuItem onClick={handleMenuClose}>Log Out</MenuItem>
      </Menu>
    </>
  )

  return (
    <div className={classes.grow}>
      <AppBar position="static" className={classes.appBar}>
        <Toolbar>
          <IconButton
            edge="start"
            className={classes.menuButton}
            color="inherit"
            aria-label="open drawer"
            onClick={openMenu('left', true)}
          >
            <MenuIcon />
          </IconButton>
          <div className={classes.growCenter}>
            <Typography className={classes.title} variant="h6" noWrap>
              <Link href='/'>
                <a>
                  <img className={classes.icon} src="/images/background/logo-d4-icon.png" alt="d4 icon" />
                  SanctuaryWorld
                </a>
              </Link>
            </Typography>
          </div>
          <div className={classes.sectionDesktop}>
            {/*<IconButton aria-label="show 4 new mails" color="inherit">*/}
            {/*  <Badge badgeContent={4} color="secondary">*/}
            {/*    <MailIcon />*/}
            {/*  </Badge>*/}
            {/*</IconButton>*/}
            {/*<IconButton aria-label="show 17 new notifications" color="inherit">*/}
            {/*  <Badge badgeContent={17} color="secondary">*/}
            {/*    <NotificationsIcon />*/}
            {/*  </Badge>*/}
            {/*</IconButton>*/}
            <IconButton
              edge="end"
              aria-label="account of current user"
              aria-controls={menuId}
              aria-haspopup="true"
              onClick={handleMenuOpen}
              color="inherit"
            >
              <AccountCircle />
            </IconButton>
          </div>
          <div className={classes.sectionMobile}>
            <IconButton
              aria-label="show more"
              aria-controls={mobileMenuId}
              aria-haspopup="true"
              onClick={handleMenuOpen}
              color="inherit"
            >
              <MoreIcon />
            </IconButton>
          </div>
        </Toolbar>
      </AppBar>
      {renderMobileMenu}
      <div className={classes.divider} />
    </div>
  )
}

ResponsiveNavBar.propTypes = {
  openMenu: PropTypes.func,
}

ResponsiveNavBar.defaultProps = {
  openMenu: () => {},
}

export default ResponsiveNavBar
