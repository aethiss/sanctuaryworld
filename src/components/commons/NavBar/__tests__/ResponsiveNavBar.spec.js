import React from 'react'
import { createMount } from '@material-ui/core/test-utils'
import ResponsiveNavBar from '../ResponsiveNavBar'

import { configure } from 'enzyme'
import Adapter from '@wojtekmaj/enzyme-adapter-react-17'

configure({ adapter: new Adapter() })

describe('ResponsiveNavBar', () => {
  it('[ResponsiveNavBar] Default render', () => {
    const renderedValue = createMount()(
      <ResponsiveNavBar openMenu={jest.fn()} />
    )
    expect(renderedValue.html()).toMatchSnapshot()
  })
})
