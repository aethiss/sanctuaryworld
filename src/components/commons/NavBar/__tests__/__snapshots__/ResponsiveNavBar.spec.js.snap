// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResponsiveNavBar [ResponsiveNavBar] Default render 1`] = `"<div class=\\"makeStyles-grow-1\\"><header class=\\"MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary makeStyles-appBar-2 MuiPaper-elevation4\\"><div class=\\"MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters\\"><button class=\\"MuiButtonBase-root MuiIconButton-root makeStyles-menuButton-4 MuiIconButton-colorInherit MuiIconButton-edgeStart\\" tabindex=\\"0\\" type=\\"button\\" aria-label=\\"open drawer\\"><span class=\\"MuiIconButton-label\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z\\"></path></svg></span><span class=\\"MuiTouchRipple-root\\"></span></button><div class=\\"makeStyles-growCenter-5\\"><h6 class=\\"MuiTypography-root makeStyles-title-6 MuiTypography-h6 MuiTypography-noWrap\\"><a href=\\"/\\"><img class=\\"makeStyles-icon-7\\" src=\\"/images/background/logo-d4-icon.png\\" alt=\\"d4 icon\\">SanctuaryWorld</a></h6></div><div class=\\"makeStyles-sectionDesktop-8\\"><button class=\\"MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeEnd\\" tabindex=\\"0\\" type=\\"button\\" aria-label=\\"account of current user\\" aria-controls=\\"primary-search-account-menu\\" aria-haspopup=\\"true\\"><span class=\\"MuiIconButton-label\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z\\"></path></svg></span><span class=\\"MuiTouchRipple-root\\"></span></button></div><div class=\\"makeStyles-sectionMobile-9\\"><button class=\\"MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit\\" tabindex=\\"0\\" type=\\"button\\" aria-label=\\"show more\\" aria-controls=\\"primary-search-account-menu-mobile\\" aria-haspopup=\\"true\\"><span class=\\"MuiIconButton-label\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z\\"></path></svg></span><span class=\\"MuiTouchRipple-root\\"></span></button></div></div></header><div class=\\"makeStyles-divider-3\\"></div></div>"`;
